backend/extractors/tram_classifier_extractor.py

```diff
@@ -0,0 +1,127 @@
+        self.model = torch.jit.load(scripted_model_path).to(self.device)
+        self.model.eval()
+
+        label_binarizer_path = hf_hub_download(repo_id=model_repo, filename="label_binarizer.pkl")
+        self.mlb = joblib.load(label_binarizer_path)
+
+        mitre_json_path = hf_hub_download(repo_id=model_repo, filename="enterprise-attack.json")
+        self.mitre_map = self._load_mitre_map(mitre_json_path)
+
+    def _load_mitre_map(self, json_path: str):
+        with open(json_path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+
+        mapping = {}
+        for obj in data["objects"]:
+            if obj.get("type") == "attack-pattern":
+                for ref in obj.get("external_references", []):
+                    if ref.get("source_name") == "mitre-attack" and "external_id" in ref:
+                        mapping[ref["external_id"]] = {
+                            "name": obj.get("name", "Unknown"),
+                            "url": ref.get("url", "")
+                        }
+        return mapping
+
+    def classify(self):
+        chunks = []
+
+        # 1. Split article
+        if len(self.tokenizer.encode(self.article_content, add_special_tokens=False)) > 512:
+            chunks = self._split_text_into_chunks(self.article_content)
+        else:
+            chunks = [self.article_content]
+
+        # 2. Add keywords
+        if self.keywords:
+            keyword_text = " ".join(self.keywords)
+            chunks.append(keyword_text)
+
+        # 3. Add QnA answers
+        for item in self.qna:
+            answer = item.get("answer", "").strip()
+            if answer:
+                if len(self.tokenizer.encode(answer, add_special_tokens=False)) > 512:
+                    chunks.extend(self._split_text_into_chunks(answer))
+                else:
+                    chunks.append(answer)
+
+        # 4. Predict
+        all_preds = {}
+        for chunk in chunks:
+            inputs = self.tokenizer(chunk, return_tensors="pt", padding=True, truncation=True, max_length=512)
+            inputs = {k: v.to(self.device) for k, v in inputs.items()}
+
+            with torch.no_grad():
+                logits = self.model(inputs["input_ids"], inputs.get("attention_mask"), inputs.get("token_type_ids"))
+                probs = torch.sigmoid(logits).cpu().numpy()[0]
+
+            for idx, prob in enumerate(probs):
+                if prob > self.threshold:
+                    label = self.mlb.classes_[idx]
+                    all_preds[label] = all_preds.get(label, 0) + prob
+
+        # Sort by descending probability
+        sorted_preds = sorted(all_preds.items(), key=lambda x: x[1], reverse=True)
+        sorted_labels = [tid for tid, _ in sorted_preds]
+        return self.enrich_ids_with_metadata(sorted_labels)
+
+    def _split_text_into_chunks(self, text, max_tokens=512, stride=256):
+        sentences = blingfire.text_to_sentences(text).split('\n')
```

Comment:
we already use `LangChain` so consider using https://pypi.org/project/langchain-text-splitters

Reply:
I mean from langchain.text_splitter